api/app/src/mappings/google/index.ts

```diff
@@ -1,36 +1,34 @@
-import { Sample } from '@metriport/api/lib/models/common/sample'
+import { Sample } from "@metriport/api/lib/models/common/sample";
 import dayjs from "dayjs";
-import convert from 'convert-units';
+import convert from "convert-units";
 
 import { GooglePoint } from "./models";
 
-export const getSamples = (arr: GooglePoint, valueIndex: number = 0): Sample[] => {
+export const getSamples = (arr: GooglePoint, valueIndex = 0): Sample[] => {
   return arr.map(item => {
     const hasFpVal = item.value.filter(val => val.fpVal);
     const startTimeNanos = Number(item.startTimeNanos);
 
     return {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       value: hasFpVal[valueIndex].fpVal!,
-      time: dayjs(convert(startTimeNanos).from('ns').to('ms')).format(
-        "YYYY-MM-DDTHH:mm:ssZ"
-      )
-    }
-  })
-}
+      time: dayjs(convert(startTimeNanos).from("ns").to("ms")).format("YYYY-MM-DDTHH:mm:ssZ"),
+    };
+  });
+};
 
 export enum ValueKey {
   fpVal = "fpVal",
-  intVal = "intVal"
+  intVal = "intVal",
 }
 
 export const getValues = (arr: GooglePoint, key: ValueKey = ValueKey.fpVal): number[] => {
   return arr.reduce((acc, curr) => {
-    const hasFpVal = curr.value.filter(val => val[key]);
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const hasFpVal = curr.value.filter(val => val[key]).map(val => val[key]!);
 
-    hasFpVal.forEach(val => {
-      acc.push(val[key]!)
-    })
+    hasFpVal.forEach(val => acc.push(val));
```

Comment:
This code is getting a bit funky now, can we just change this whole bit to something like:

``` typescript
    curr.value.forEach(val => {
      const fpVal = val[key];
      if (fpVal) acc.push(fpVal);
    });
```

Reply:
^ @Orta21